internal/charmstore/store_test.go

```diff
@@ -1119,6 +1126,15 @@ func (s *StoreSuite) TestAddUserOwnedCharmArchive(c *gc.C) {
 	s.checkAddCharm(c, charmArchive, false, newResolvedURL("~charmers/precise/wordpress-1", -1))
 }
 
+func (s *StoreSuite) TestAddDevelopmentCharmArchive(c *gc.C) {
+	charmArchive := storetesting.Charms.CharmArchive(c.MkDir(), "wordpress")
+	url := newResolvedURL("~charmers/precise/wordpress-1", 1)
```

Comment:
perhaps add a boolean to newResolvedURL for development as that function is only used for tests, although it's used in lots of them.